lib/src/cli.rs

```diff
@@ -451,22 +493,20 @@ async fn container_store(
     proxyopts: ContainerProxyOpts,
 ) -> Result<()> {
     let mut imp = ImageImporter::new(repo, imgref, proxyopts.into()).await?;
+    let layer_progress = imp.request_progress();
     let prep = match imp.prepare().await? {
         PrepareResult::AlreadyPresent(c) => {
             println!("No changes in {} => {}", imgref, c.merge_commit);
             return Ok(());
         }
         PrepareResult::Ready(r) => r,
     };
-    for layer in prep.all_layers() {
-        if layer.commit.is_some() {
-            println!("Using layer: {}", layer.digest());
-        } else {
-            let size = crate::glib::format_size(layer.size());
-            println!("Downloading layer: {} ({})", layer.digest(), size);
-        }
-    }
-    let import = imp.import(prep).await?;
+    print_layer_status(&prep);
+    let progress_printer =
+        tokio::task::spawn(async move { handle_layer_progress_print(layer_progress).await });
```

Comment:
Incidentally I think what we want here is https://blog.yoshuawuyts.com/postfix-spawn/ to lifecycle-bind this task to the parent one.  Will look at that at some point.  Maybe https://lib.rs/crates/async_nursery ?